File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/request/type/CreatedSkillProvisionImpulse.kt
package gaia.sdk.request.type

import gaia.sdk.client.Type
import gaia.sdk.request.intf.*
import gaia.sdk.client.Input
import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.request.input.*
import gaia.sdk.request.enumeration.*

/**
 * Impulse which indicates the result of a create SkillProvision impulse
 */
class CreatedSkillProvisionImpulse: Type() {

    fun id() { 
        add {"id" } 
    }

    /**
     * the SkillProvision instance
     */
    fun data(config: SkillProvision.() -> Unit) = 
        add { "data{ " + SkillProvision().apply(config).render(it) + "}"}

}

File: gaia-sdk-javascript/src/graphql/request/type/CreatedSkillProvisionImpulse.ts

import {SkillProvision} from "./SkillProvision";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Impulse which indicates the result of a create SkillProvision impulse
 */
export class CreatedSkillProvisionImpulse extends Array<(_:VariableRegistry) => string> {
public _typeName = "CreatedSkillProvisionImpulse";
    public id = () => { 
        this.push(_ => "id")
    };

    /**
     * the SkillProvision instance
     */
    public data = (config: (_:SkillProvision) => void) => this.push((registry) => {
        const entity = new SkillProvision();
        config(entity);
        return "data { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

